prism/src/main/java/org/apache/falcon/resource/AbstractExtensionManager.java

```diff
@@ -99,13 +100,21 @@ public APIResult getExtensions() {
         }
     }
 
-    public APIResult deleteExtensionMetadata(String extensionName){
+    public APIResult deleteExtensionMetadata(String extensionName) throws FalconException {
         validateExtensionName(extensionName);
-        try {
-            return new APIResult(APIResult.Status.SUCCEEDED, ExtensionStore.get().deleteExtension(extensionName,
-                    CurrentUser.getUser()));
-        } catch (Throwable e) {
-            throw FalconWebException.newAPIException(e, Response.Status.INTERNAL_SERVER_ERROR);
+        return new APIResult(APIResult.Status.SUCCEEDED, deleteExtension(extensionName));
+    }
+
+    private String deleteExtension(String extensionName) throws FalconException {
```

Comment:
For better readability and consistency with rest of the methods in this class, can we call this checkNoJobsExist and keep store deletion in the deleteExtensionMetadata itself?